client/apps/edit/components/section_text/index.coffee

```diff
@@ -1,145 +1,397 @@
-        li: true
+  focus: (e) ->
+    @setState focus: true
+    @refs.editor.focus()
+
+  onBlur: ->
+    @setState focus: false
+
+  convertFromHTML: (html) ->
+    blocksFromHTML = convertFromHTML({
+      htmlToStyle: (nodeName, node, currentStyle) ->
+        if nodeName is 'span' and node.style.textDecoration is 'line-through'
+          return currentStyle.add 'STRIKETHROUGH'
+        else
+          return currentStyle
+      htmlToEntity: (nodeName, node) ->
+        if nodeName is 'a'
+          data = {url: node.href, name: node.name, className: node.classList.toString()}
+          return Entity.create(
+              'LINK',
+              'MUTABLE',
+              data
+          )
+        if nodeName is 'p' and node.innerHTML is '<br>'
+          node.innerHTML = '' # remove <br>, it renders extra breaks in editor
+      })(html)
+    return blocksFromHTML
+
+  convertToHtml: (editorState) ->
+    html = convertToHTML({
+      entityToHTML: (entity, originalText) ->
+        if entity.type is 'LINK'
+          sanitizeName = originalText.split(' ')[0].replace(/[.,\/#!$%\^&\*;:{}=\-_`’'~()]/g,"")
+          name = if entity.data.name then ' name="' + sanitizeName + '"' else ''
+          if entity.data.className.includes('is-follow-link')
+            return '<a href="' + entity.data.url + '" class="' + entity.data.className + '"' + name + '>' + originalText + '</a><a class="entity-follow artist-follow"></a>'
+          else if entity.data.className is 'is-jump-link'
+            return a { name: sanitizeName, className: entity.data.className}
+          else
+            return a { href: entity.data.url}
+        return originalText
+      blockToHTML: (block) ->
+        if block.type is 'header-three'
+          return h3 {}, block.text
+      styleToHTML: (style) ->
+        if style is 'STRIKETHROUGH'
+          return span { style: {textDecoration: 'line-through'}}
+    })(editorState.getCurrentContent())
+    # put the line breaks back for correct client rendering
+    html = html.replace('<p></p>', '<p><br></p>').replace('<p> </p>', '<p><br></p>')
+    return html
+
+  stripGoogleStyles: (html) ->
+    # remove non-breaking spaces between paragraphs
+    html = html.replace('</p><br>', '</p>').replace('<br class="Apple-interchange-newline">', '')
+    doc = document.createElement('div')
+    doc.innerHTML = html
+    # remove dummy b tags google docs wraps document in
+    boldBlocks = doc.getElementsByTagName('B')
+    for block, i in boldBlocks
+      if block.style.fontWeight is 'normal'
+        $(doc.getElementsByTagName('B')[i]).replaceWith(doc.getElementsByTagName('B')[i].innerHTML)
+    # replace bold spans with actual b tags
+    boldSpans = doc.getElementsByTagName('SPAN')
+    for span, i in boldSpans
+      if span?.style.fontWeight is '700'
+        newSpan = '<strong>' + span.innerHTML + '</strong>'
+        $(doc.getElementsByTagName('SPAN')[i]).replaceWith(newSpan)
+    return doc.innerHTML
```

Comment:
Google Docs styles: so ugly :'( 

Nice that we can manage this in Positron now tho, I always did find it annoying to go into the Scribe plugin to make modifications.